### ISSUE TITLE
does fmpp command line support stdout mode insteadof create file? 

### ISSUE THREAD

**Issue by leeluolee** [freemarker/fmpp#13]
I tried fmpp like 

```shell
fmpp  test/template/index.ftl   >  todo.css
```

but got following result

```shell
Failed!
The "outputRoot" setting was not set. FMPP can't start working without that.
```

I think , the `outputRoot` must a Optional arguments then developer can use stdout in other way . like text processing. 

Just a recommendation

**Comment by ddekany** [comment]
Yes, that's on the FMPP TODO actually. (It's not entirely trivial, because FMPP can output to multiple files and such, which doesn't make sense with stdout.)

**Comment by leeluolee** [comment]
I used fmpp to render a 【freemarker template】 in nodejs application.   The idealy way to use fmpp is 

```js
shell.run("fmpp some/path/to/index.ftl -D xxx", function(err, compileContent){  // data is the stdout's data stream
   response.send(compileContent)
})
```

but I found that, before I got the `compileContent`, I have to 【`read from a tmp file`】 then 【`unlink the tmp file after reading`】.  (in fmpp process, there is also a useless writing operation)

**Comment by ddekany** [comment]
How it probably will be, if I ever had time for it, is that you can pass `--stdin` and/or `--stdout` to FMPP. This also means that one can combine the two and use FMPP as a filter. Why I don't want these to be the default is simply that the typical usage patterns is file-to-file and directory-to-directory, and if there will be default then FMPP can give less useful error messages what that's messed up. User will just see the screen flooded or FMPP being frozen (actually waiting for stdin)...

**Comment by leeluolee** [comment]
`--stdin` or `--stdout` is good enough for my needs，  thanks .  Looking forward to your updating.  :)

**Comment by katiesandford** [comment]
:+1: When will this be released? :)

**Comment by ddekany** [comment]
It wasn't even implemented, so I really don't know.

**Comment by MulixBF** [comment]
Trivial kludge in bash. Maybe someone would be interested

function freemaker() {
    template_file=$(mktemp)
    output_file=$(mktemp)
    template=$1 && shift

    echo "$template" > "$template_file"
    fmpp -q "$template_file" -o "$output_file" $@
    cat "$output_file"
    rm "$template_file"
    rm "$output_file"
}
